source/Coop/Mod/Managers/ClientCharacterCreatorManager.cs

```diff
@@ -78,16 +82,22 @@ public override void OnLoadFinished()
         private void SkipCharacterCreation()
         {
             CharacterCreationState characterCreationState = GameStateManager.Current.ActiveState as CharacterCreationState;
-            bool flag = CharacterObject.PlayerCharacter.Culture == null;
+            bool flag = characterCreationState.CurrentStage is CharacterCreationCultureStage;
             if (flag)
             {
-                CultureObject culture = CharacterCreationContentBase.Instance.GetCultures().FirstOrDefault<CultureObject>();
+                CultureObject culture = CharacterCreationContentBase.Instance.GetCultures().GetRandomElementInefficiently();
                 CharacterCreationContentBase.Instance.SetSelectedCulture(culture, characterCreationState.CharacterCreation);
                 characterCreationState.NextStage();
             }
             bool flag2 = characterCreationState.CurrentStage is CharacterCreationFaceGeneratorStage;
             if (flag2)
             {
+                CharacterCreationFaceGeneratorView stageView = characterCreationState.CurrentStage.Listener as CharacterCreationFaceGeneratorView;
+                BodyGeneratorView bgv = (BodyGeneratorView)stageView.GetType().GetField("_faceGeneratorView", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(stageView);

```

Comment:
Why cast stageView as a CharacterCreationFaceGeneratorView if we're just going to coerce it as a BodyGeneratorView to reflect some things?

Reply:
Good point